modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/utils/getFields.ts

```diff
@@ -108,7 +122,7 @@ function getValidFields({
 
 				if (parentSchemaName) {
 					field.name = `${field.name}${FDS_NESTED_FIELD_NAME_PARENT_SUFFIX}`;
-					field.type = schemas[parentSchemaName]?.type || 'object';
+					field.type = schemas[parentSchemaName]?.type ?? 'object';
 					targetSchemaName = parentSchemaName;
```

Comment:
I'm not 100% sure, so this is more a question than a change request.

When `targetSchemaName` changes, I believe we need to update `xFilterable` array with the corresponding fields. Iinm, each schema defines the filterables, so, that array shall contain only the fields of the schema we are using to call `getValidFields` isn't it?

Current implementation is using all the time the filterables from the original schema (see line 187). I'm not sure this is what we want when examining a connected schema via `$ref ` (line 98) or via `x-parent-map` (line 112). Are the fields in these connected schemas included in the original schema too? If that is the case, then current impl seems correct.